packages/instant-apps-components/src/components/instant-apps-header/instant-apps-header.test.ts

```diff
@@ -0,0 +1,64 @@
+import { expect, test } from 'vitest';
+
+import '../../../dist/components/instant-apps-header.js';
+const header = document.createElement('instant-apps-header');
+
+test('should render component with custom title text', async () => {
+  const customTitle = 'Instant Apps Header';
+  header.titleText = customTitle;
+  document.body.appendChild(header);
+  await new Promise(resolve => requestIdleCallback(resolve));
+  
+  expect(header.titleText).toBe(customTitle);
+});
+
+test('test info button visibility', async () => {
+  header.infoButton = true;
+  header.infoTitleText = "info test"
+  document.body.appendChild(header);
+  await new Promise(resolve => requestIdleCallback(resolve));
+  header.toggleInfo();
+  expect(header.infoButton).toBe(true);
+  expect(header.infoIsOpen).toBe(true);
+  header.toggleInfo();
+  expect(header.infoIsOpen).toBe(false);
+  const infoElement = header.shadowRoot?.querySelector('#infoButton');
+  expect(infoElement? true : false).toBe(true);
+  const infosTitle = infoElement?.getAttribute('title')
+  expect(infosTitle).toBe(header.infoTitleText);
+})
+
+
+test('logo scale change should be reflected', async () => {
+  header.logoScale = 's';
+  document.body.appendChild(header);
+  await new Promise(resolve => requestIdleCallback(resolve));
+  expect(header.logoScale).toBe('s');
+
+})
+
+test('logo scale change to large should be reflected', async () => {
+  header.logoScale = 'l';
+  document.body.appendChild(header);
+  await new Promise(resolve => requestIdleCallback(resolve));
+  expect(header.logoScale).toBe('l');
```

Comment:
maybe query for the node and check for it's scale value

Reply:
applied in other places. Try querying for the nodes and testing the values within the node elements.

Reply:
once the component dom is rendered, check for a way to ensure that the logoScale that you set is being applied. like querying for a dom node that is using the value you set i.e. `l`.